src/hooks/useGame.tsx

```diff
@@ -1,134 +1,172 @@
 import { useCallback, useState } from 'react';
 
-import useGameBoard from './useGameBoard';
-import useMoveLeft from './useMoveLeft';
-import useRotateCells from './useRotateCells';
-
-export enum GameOverStatus {
-  Success = 'success',
-  Fail = 'fail',
-  None = 'none',
-}
+import type {
+  CellsType,
+  DirectionType,
+  HistoryType,
+} from '../components/types/GameType';
+import { GameOverStatus } from '../components/types/GameType';
+import { addOneRandomCell, addTwoRandomCells } from '../utils/addRandomCell';
+import { checkCanMove, is128Exist } from '../utils/checkGameOver';
+import { getHighScore, getScore } from '../utils/getScore';
+import {
+  getLocalData,
+  resetLocalStorage,
+  saveLocalStorage,
+} from '../utils/localStorage';
+import moveCells from '../utils/moveCells';
+import moveCellsByDirection from '../utils/moveCellsByDirection';
+import undo from '../utils/undo';
 
 const useGame = () => {
-  const {
-    isMoved,
-    cells,
-    history,
-    score,
-    highScore,
-    getEmptyCellsIndex,
-    setCells,
-    setIsMoved,
-    addTwoRandomCells,
-    saveCellsHistory,
-    undo,
-    getScore,
-    updateHighScore,
-    initGameBoard,
-  } = useGameBoard();
-  const { rotateMapCounterClockwise } = useRotateCells();
-  const { moveLeft } = useMoveLeft();
-
+  const [cells, setCells] = useState<CellsType>(
+    Array<null>(4)
+      .fill(null)
+      .map(() => Array<null>(4).fill(null)),
+  );
+  const [history, setHistory] = useState<HistoryType>([]);
+  const [score, setScore] = useState<number>(0);
+  const [highScore, setHighScore] = useState<number>(0);
```

Comment:
```suggestion
  const [score, setScore] = useState(0);
  const [highScore, setHighScore] = useState(0);
```

number string 이런 건 자동추론이 될 경우 안 써주는 게 더 낫습니다!